codec-http2/src/test/java/io/netty/handler/codec/http2/Http2MultiplexCodecTest.java

```diff
@@ -0,0 +1,265 @@
+            .status(new AsciiString("200"));
+
+    @Before
+    public void setUp() throws Exception {
+        channel.connect(null);
+        channel.pipeline().addLast(serverCodec);
+        http2HandlerCtx = channel.pipeline().context(serverCodec.connectionHandler());
+        serverCodecCtx = channel.pipeline().context(serverCodec);
+
+        // Handshake
+        verify(frameWriter).writeSettings(eq(http2HandlerCtx),
+            anyHttp2Settings(), anyChannelPromise());
+        verifyNoMoreInteractions(frameWriter);
+        channel.writeInbound(Http2CodecUtil.connectionPrefaceBuf());
+        frameListener.onSettingsRead(http2HandlerCtx, new Http2Settings());
+        verify(frameWriter).writeSettingsAck(eq(http2HandlerCtx), anyChannelPromise());
+        frameListener.onSettingsAckRead(http2HandlerCtx);
+    }
+
+    @Test
+    public void startStop() throws Exception {
+        assertTrue(channel.isActive());
+        channel.close();
+        verify(frameWriter).writeGoAway(
+                eq(http2HandlerCtx), eq(0), eq(0L), eq(Unpooled.EMPTY_BUFFER), anyChannelPromise());
+        assertTrue(!channel.isActive());
+    }
+
+    @Test
+    public void headerRequestHeaderResponse() throws Exception {
+        LastInboundHandler stream = new LastInboundHandler();
+        streamInit.handler = stream;
+        frameListener.onHeadersRead(http2HandlerCtx, 1, request, 31, true);
+        assertNull(streamInit.handler);
+        assertEquals(new DefaultHttp2HeadersFrame(request, true, 31), stream.readInbound());
+        assertNull(stream.readInbound());
+        assertTrue(stream.channel().isActive());
+
+        stream.writeOutbound(new DefaultHttp2HeadersFrame(response, true, 27));
+        verify(frameWriter).writeHeaders(
+                eq(http2HandlerCtx), eq(1), eq(response), anyInt(), anyShort(), anyBoolean(),
+                eq(27), eq(true), anyChannelPromise());
+        verify(frameWriter, never()).writeRstStream(
+            any(ChannelHandlerContext.class), anyInt(), anyLong(), anyChannelPromise());
+        assertFalse(stream.channel().isActive());
+        assertTrue(channel.isActive());
+    }
+
+    @Test
+    public void entityRequestEntityResponse() throws Exception {
+        LastInboundHandler stream = new LastInboundHandler();
+        streamInit.handler = stream;
+        frameListener.onHeadersRead(http2HandlerCtx, 1, request, 0, false);
+        assertEquals(new DefaultHttp2HeadersFrame(request, false), stream.readInbound());
+        assertNull(stream.readInbound());
+        assertTrue(stream.channel().isActive());
+
+        ByteBuf hello = bb("hello");
+        frameListener.onDataRead(http2HandlerCtx, 1, hello, 31, true);
+        // Release hello to emulate ByteToMessageDecoder
+        hello.release();
+        Http2DataFrame inboundData = stream.readInbound();
+        assertEquals(new DefaultHttp2DataFrame(bb("hello"), true, 31), inboundData);
```

Comment:
So it looks like the answer is "Yes, it needs to be". Based on teamcity resuts it looks like unpooled bytebufs are noticed if they leak. I'm now releasing it.